cranelift/isle/veri/veri_engine/src/solver.rs

```diff
@@ -25,6 +26,48 @@ use encoded_ops::subs;
 
 use crate::MAX_WIDTH;
 
+// Verification state, used by vir_expr_to_sexp
+#[derive(Clone)]
+pub struct State {
+    val: SExpr,
+
+    load_args: Option<Vec<SExpr>>,
+    store_args: Option<Vec<SExpr>>,
+}
+
+pub fn val_state(val: SExpr) -> State {
+    State {
+        val,
+        load_args: None,
+        store_args: None,
+    }
+}
+
+// Merge states assuming there is zero or one "some" value for each field
+pub fn merge_states_assert_lone(val: SExpr, states: Vec<State>) -> State {
+    // If load_args is always none, set the new state's load_args to be none
+    // If load_args is some in one spot, set the new state's load_args to be that
+    // Otherwise, panic
+
+    // Same logic for store_args
+    let mut new_state = val_state(val);
+    let mut set_load_arg = None;  
+    for state in states.iter() {
+         if set_load_arg.is_none() && state.load_args.is_some() {
```

Comment:
It would be more Rust style to implement these checks with a `match` statement instead of a series of if statements